### PR TITLE
DOC: document the release process

### PR DESCRIPTION
**Pull request by llllllllll**
Typo hunting begins now!

### CODE REVIEW

File: docs/source/release-process.rst [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+
+Updating the Release Notes
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When we are ready to ship a new release of zipline, edit the :doc:`releases`
+page. We will have been maintaining a whatsnew file while working on the release
+with the new version. First, find that file in:
+``docs/source/whatsnew/<version>.txt``. It will be the highest version number.
+Edit the release date field to be todays date in the format:
```
Comment (richafrank) [inline docs/source/release-process.rst]: "today's"

File: docs/source/release-process.rst [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+   .. include:: whatsnew/<version>.txt
+
+
+Updating the ``__version__``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We use `versioneer <https://github.com/warner/python-versioneer>`__ to
+manage the ``__version__`` and ``setup.py`` version. This means that we pull
+this information from our version control's tags to ensure that they stay in
+sync and to have very fine grain version strings for development installs.
```
Comment (richafrank) [inline docs/source/release-process.rst]: Is it "fine grain" or "fine grained"?

File: docs/source/release-process.rst [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+.. code-block:: bash
+
+   $ pip install <zipline-root>/dist/zipline-<major>.<minor>.<micro>.tar.gz
+   $ python -c 'import zipline;print(zipline.__version__)'
+
+This should print the version we are expecting to release.
+
+.. note::
+
+   It is very important to both ``cd`` into a clean directory and make a clean
```
Comment (richafrank) [inline docs/source/release-process.rst]: :+1:

File: docs/source/release-process.rst [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+.. code-block:: bash
+
+   $ python setup.py sdist upload -r pypitest
+
+
+.. note::
+
+   If the package version has been taken: locally update your setup.py to
+   override the version with a new number. Do not use the next version, just
+   append a ``.<nano>`` section to the current version. PyPI prevents the same
```
Comment (richafrank) [inline docs/source/release-process.rst]: Would it be easier to delete that version from the test server?

File: docs/source/release-process.rst [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+This should pull the package you just uploaded and then print the version
+number.
+
+Now that we have tested locally and on PyPI test, it is time to upload to PyPI:
+
+.. code-block:: bash
+
+   $ python setup.py sdist upload
+   $ python -c 'import zipline;print(zipline.__version__)'
```
Comment (richafrank) [inline docs/source/release-process.rst]: Should these two be run in different virtualenvs?

File: docs/source/release-process.rst [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+First, build the wheels locally with:
+
+.. code-block:: bash
+
+   $ python setup.py bdist_wheel
+
+
+Just like the ``sdist``, we need to ``cd`` into a clean directory and use a
+clean virtualenv. Then, test that the wheel was built successfully with:
```
Comment (richafrank) [inline docs/source/release-process.rst]: Should we mention something about building for each version of python?

File: docs/source/release-process.rst (hunk 4 above)
Comment (llllllllll) [inline docs/source/release-process.rst]: You can still never use it again, even if it has been deleted

File: docs/source/release-process.rst (hunk 5 above)
Comment (llllllllll) [inline docs/source/release-process.rst]: yeah, I will split these commands

File: docs/source/release-process.rst (hunk 4 above)
Comment (richafrank) [inline docs/source/release-process.rst]: Oh wow, ok.

File: docs/source/release-process.rst (hunk 6 above)
Comment (llllllllll) [inline docs/source/release-process.rst]: done

**Pull request merged**